framework/src/engine/consensus/fork_choice/fork_choice_rule.ts

```diff
@@ -24,13 +24,23 @@ export enum ForkStatus {
 	DISCARD = 6,
 }
 
+interface Slots {
+	getSlotTime(slot: number): number;
+	getSlotNumber(timeStamp: number): number;
+	isWithinTimeslot(slot: number, time: number): boolean;
+}
+
 export const forgingSlot = (slots: Slots, block: BFTHeader): number =>
 	slots.getSlotNumber(block.timestamp);
 
 export const isBlockReceivedWithinForgingSlot = (
 	slots: Slots,
 	{ timestamp, receivedAt }: BFTHeader,
-): boolean => slots.isWithinTimeslot(slots.getSlotNumber(timestamp), receivedAt);
+): boolean =>
+	slots.isWithinTimeslot(
+		slots.getSlotNumber(timestamp),
+		receivedAt ?? Math.floor(Date.now() / 1000),
```

Comment:
`Math.floor(Date.now() / 1000)`
Does the default value come from a LIP ?

Reply:
I think this was default value in the implementation from the past.

receivedAt is defined only when received from the P2P, and if not defined, it just treat as it is just received now

Reply:
ok, probably the confusion was since there is no default value for `receivedAt` in the current implementation, do you know why ?

Reply:
it was internally handled in the `isWithinTimeslot` / Slot before, but in the LIP method, there is no optional so it's handled outside